src/Serialization/basic_types.jl

```diff
@@ -1,5 +1,7 @@
 ################################################################################
 # fmpz
+registerSerializationType(fmpz, "fmpz")
```

Comment:
So the idea of this new `registerSerializationType` helper is to decouple things a bit more: the goal is that adding serialization to a new type should be possible by adding stuff in a single place, and not require modifying the initialization code of some global object. As usual, being modular has clear benefits, e.g. that would make it possible for a package to extend serialization to additional types.

(We may want to eventually allow doing more things, like specifying a namespace override, but one step after the other...)

Reply:
This looks great, and your able to get around the Revise issue with this? I am also just wondering if this works because the typeMaps are declared as constants

Reply:
This helper is only secondary to making it work with `Revise`. Rather, the important bit is not initialize the bulk of the typeMap via an initial value, because editing that initial value then leads to problems, as Revise can't replace the initial dict that was created, it only can create a new one.

So instead we created an *empty* initial dict (so we never have to edit that initialization, and hence `Revise` never has to, well, revise it), and fill it with content separately. In the first commit of this PR, this is done by a simple loop. Then in the next commit, this helper function is added to "modularize" it.